azurerm/internal/services/cognitive/resource_arm_cognitive_account.go

```diff
@@ -146,11 +163,16 @@ func resourceArmCognitiveAccountCreate(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("error expanding sku_name for Cognitive Account %s (Resource Group %q): %v", name, resourceGroup, err)
 	}
 
+	accountProperties := expandCognitiveAccountProperties(d)
+	if err := validateCognitiveAccountProperties(accountProperties, kind); err != nil {
```

Comment:
We should just do the validation in the expand function to be consistent with the rest of the provider

Reply:
done